Backend/implementing logging

DanielNikolic03 opened the pull request:
This branch has implemented logging and slight clean up

DanielNikolic03 committed:
Added log table

DanielNikolic03 committed:
Modifying logging table in DB

The initial thought of accessing the logging table from the data layer would not work efficiently, as a reference from the access layer to data layer would be required. This commit removes the data layer accessing and slightly improves the tables attributes.

DanielNikolic03 committed:
Creation of Meta-TV2-Utils

Meta-TV2-Utils will act as a layer referenced by all other layers. As of right now, Meta-TV2-Utils will only contain utility for logging, but could support more functionality.

DanielNikolic03 committed:
Merge branch 'dev' into backend/implementing-logging

DanielNikolic03 committed:
Added logging to datalayer

DanielNikolic03 committed:
Fixed so groups and slides cant be archived multiple times

DanielNikolic03 committed:
Merge branch 'dev' into backend/implementing-logging

DanielNikolic03 committed:
Logging complete for datalayer

DanielNikolic03 committed:
Added some more informative loggs in datalayer

DanielNikolic03 committed:
Added debug logging in ApiController

jonathan-blomlof reviewed (approved):
Looks good! However I believe when logging errors from `try` blocks, the exceptions will be because the connection to the database isn't working. Thus it will not be possible to log the error either. But I like the start and suggest to modify the error loggings to write to stderr or something.

DanielNikolic03 committed:
Changed logging in datalayer to simple console output

As stated by Jonathans comment, the only time an exception would be thrown in the datalayer would be when there are issues with the database. That means logging wouldn't work either

DanielNikolic03 committed:
Adding exception logging in businesslayer

Putting try blocks around code that can throw, and adding logging in those blocks
